test/stat/test_stat.c

```diff
@@ -193,6 +193,51 @@ void test() {
   err = stat("folder/subdir", &s);
   assert(s.st_mtime != TEST_TIME);
 
+  EM_ASM(
+    var stats = FS.stat("folder/file");
+    assert(stats.dev == 1);
+    assert(stats.ino);
+    assert(stats.mode == 33279);
+    assert(stats.nlink);
+    assert(stats.rdev == 0);
+    assert(stats.size == 6);
+    assert(stats.atime);
+    assert(stats.mtime);
+    assert(stats.ctime);
+    var original_mode = stats.mode;
+
+    FS.symlink("file", "folder/symlinkfile");
+    FS.chmod("folder/file", 0o000);
+
+    var linkStats = FS.lstat("folder/symlinkfile");
+    assert(linkStats.dev == 1);
+    assert(linkStats.ino);
+    assert(linkStats.mode != original_mode);
```

Comment:
Can't we assert on the specific new mode's value?

Reply:
I see your point. Actually, I noticed that the legacy JS API and WasmFS create symlinks with different permissions. Although, I read that in POSIX symlink permissions are effectively ignored, so would this be an issue?

Reply:
What permissions do they create with? Maybe the WasmFS permissions are wrong, and we need to fix that, or maybe we can ifdef on the different values in the test if both are ok.

Reply:
I think POSIX ignores symlink permissions normally, but not when operating on the symlink itself (using the special functions for that), so the permissions can matter I think.

Reply:
Yes, it's the permissions on creation. So those are either `0000` or `0777`, interesting. I'm trying to really understand the issue but the docs are somewhat unclear to me; this link says the permissions don't matter but because they are effectively `0777`:

https://man7.org/linux/man-pages/man7/symlink.7.html

Anyhow, if we're not sure then it might best to make the permissions `0777` as well. But we don't need to do that in this PR, let's think about it later.